packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -508,37 +516,54 @@ private struct ImageUtils {
     return exif
   }
 
-  static func getGifDataFrom(image: UIImage,
-                             compressionQuality quality: Double?,
-                             initialMetadata: [String: Any]?) throws -> Data? {
-    guard let data = image.jpegData(compressionQuality: quality ?? DEFAULT_QUALITY) else {
-      throw FailedToReadImageDataException()
+  static func processGifData(
+    inputData: Data?,
+    compressionQuality quality: Double?,
+    initialMetadata: [String: Any]?,
+    cropRect: CGRect? = nil
+  ) throws -> Data? {
+    // for uncropped, maximum quality image we can just pass through the raw data
+    if cropRect == nil,
+       quality == nil || quality == MAXIMUM_QUALITY {
```

Comment:
Just to make it idiot-proof 😅
```suggestion
       quality == nil || quality >= MAXIMUM_QUALITY {
```